wallet/invoices.c

```diff
@@ -89,13 +93,96 @@ struct invoices *invoices_new(const tal_t *ctx,
 	return invs;
 }
 
+static void install_expiration_timer(struct invoices *invoices,
+				     struct invoice *i);
+static void trigger_expiration(struct invoice *i)
+{
+	struct invoices *invoices = i->owner;
+	u64 now = time_now().ts.tv_sec;
+	sqlite3_stmt *stmt;
+	struct invoice_waiter *w;
+
+	assert(i->state == UNPAID);
+
+	/* Timer already triggered, destroy the timer object. */
+	i->expiration_timer = tal_free(i->expiration_timer);
```

Comment:
This is technically unnecessary, since timers free themselves iff you don't.  But it's harmless.

Reply:
`install_expiration_timer` asserts that `expiration_timer` is NULL, and we can call `install_expiration_timer` later, hence the early clear.

Reply:
Yes, but we don't technically need to free it.